### PR TITLE
Create TFRecords for tracking datasets

### PR DESCRIPTION
**Pull request by vanvalen**
## What
* Added functionality to create TFRecords for tracking datasets

## Why
* As the training datasets grow in size, they are no longer able to fit in memory (as is the case with image generators). Adding functionality for TFRecords will let us train on larger datasets as they are loaded dynamically from disk during training rather than into memory all at once.

### CODE REVIEW

**Review by msschwartz21 — CHANGES_REQUESTED** [review]
Please remove the uses of `with tf.device('/cpu:0'):`. I think it's reasonable to leave a comment in the docstring that it might be necessary to assign data to the cpu, but I think it should be at the discretion of the user.

File: deepcell/utils/tfrecord_utils.py [hunk 1]
```diff
@@ -195,28 +201,255 @@ def parse_segmentation_example(example, dataset_ndims=None,
...
+            shape_string = key + '_shape_' + str(i)
+            data[shape_string] = _int64_feature(shapes[i])
+
+    # Create an Example, wrapping the single features
+    example = tf.train.Example(features=tf.train.Features(feature=data))
+
+    return example
+
+
+def write_tracking_dataset_to_tfr(track,
```
[the remark this answers is not in the document]
Comment (vanvalen) [reply deepcell/utils/tfrecord_utils.py:306]: Added

File: deepcell/utils/tfrecord_utils.py [hunk 2]
```diff
@@ -195,28 +201,255 @@ def parse_segmentation_example(example, dataset_ndims=None,
...
+                                  target_max_cells=168,
```
[the remark this answers is not in the document]
Comment (vanvalen) [reply deepcell/utils/tfrecord_utils.py:308]: Done

**Comment by vanvalen** [comment]
Addressed all of the issues raised.

**Review by msschwartz21 — APPROVED** [review]

**Pull request merged**